File: Backend/Frontend/src/app/models/objectif.ts

import { Entreprise } from "./Entreprise/entreprise";
import { Activity } from "./activity";

export class Objectif {
    id_objectif!: number;
  name_objectif!: string;
  descr_objectif!: string;
  date_debut_objectif!: Date;
  date_fin_objectif!: Date;
  type_objectif!: string;
  status_objectif!: string;
  entreprise!: Entreprise;
  activities!: Activity[];
}

File: Backend/business-server/src/main/java/tn/esprit/entities/Objectif.java
package tn.esprit.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.web.bind.annotation.CrossOrigin;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@CrossOrigin
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Objectif {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_objectif;
    private String name_objectif;
    private String descr_objectif;

    @Temporal(TemporalType.DATE)
    private Date date_debut_objectif;
    @Temporal(TemporalType.DATE)
    private Date date_fin_objectif;

    @Enumerated(EnumType.STRING)
    private Type type_objectif;
    @Enumerated(EnumType.STRING)
    private Status status_objectif = Status.PENDING;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "identreprise")
    private Entreprise entreprise;

    @OneToMany(mappedBy = "objectif")
    @JsonIgnore
    private List<Activity> activities;

}


